ast/compile.go

```diff
@@ -1726,11 +1741,18 @@ func buildComprehensionIndices(arity func(Ref) int, candidates VarSet, node inte
 	return n
 }
 
-func getComprehensionIndex(arity func(Ref) int, candidates VarSet, expr *Expr) *ComprehensionIndex {
+func getComprehensionIndex(dbg debug.Debug, arity func(Ref) int, candidates VarSet, expr *Expr) *ComprehensionIndex {
 
 	// Ignore everything except <var> = <comprehension> expressions. Extract
 	// the comprehension term from the expression.
-	if !expr.IsEquality() || expr.Negated || len(expr.With) > 0 {
+	if !expr.IsEquality() {
+		dbg.Printf("%s: not equality", expr)
+		return nil
+	} else if expr.Negated {
+		dbg.Printf("%s: negated expression", expr)
+		return nil
+	} else if len(expr.With) > 0 {
+		dbg.Printf("%s: with expression", expr)
```

Comment:
These cases are going to create a lot of noise in the debug output and I don't know if they'll be very helpful. I'd be inclined to remove these--they're similar to the `term == nil` case below.